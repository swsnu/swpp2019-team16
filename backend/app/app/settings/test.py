from .base import *
import environ
from redis import ConnectionPool

env = environ.Env()
env.read_env('env/.env.development')

DEBUG = True

# Database
# https://docs.djangoproject.com/en/2.1/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

REDIS_CONNECTION_POOL = ConnectionPool(host=env('REDIS_HOST'), port=6379, db=0)
