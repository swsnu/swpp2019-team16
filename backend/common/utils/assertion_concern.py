class AssertionConcern:

    def assert_not_none(self, val, msg):
        if val is None:
            raise TypeError(msg)
