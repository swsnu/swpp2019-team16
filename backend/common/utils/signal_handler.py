import asyncio
import functools
import signal


def register_signal_handler(loop, shutdown):
    for signame in ('SIGINT', 'SIGTERM'):
        loop.add_signal_handler(
            getattr(signal, signame),
            functools.partial(asyncio.ensure_future,
                              shutdown(signame, loop)))


async def shutdown_process(self, sig, loop):
    print('caught {0}'.format(sig))
    tasks = [task for task in asyncio.Task.all_tasks() if task is not
             asyncio.tasks.Task.current_task()]

    list(map(lambda task: task.cancel(), tasks))
    results = await asyncio.gather(*tasks, return_exceptions=True)

    print('finished awaiting cancelled tasks, results: {0}'
          .format(results))
    loop.stop()
