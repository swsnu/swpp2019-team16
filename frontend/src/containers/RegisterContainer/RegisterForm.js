import React, { useCallback, useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Register from '../../components/Register/Register';
import { register, changeField, initializeForm } from '../../modules/auth/auth';
import { check } from '../../modules/user/user';
import CircularProgress from '@material-ui/core/CircularProgress';
import { makeStyles } from '@material-ui/styles';
import theme from '../../lib/styles/theme';

const useStyles = makeStyles({
  root: {
    position: 'absolute',
    left: 0,
    top: 0,
    bottom: 0,
    right: 0,
    backgroundColor: theme.palette.primary.main,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  container: {
    marginTop: '25px',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

function RegisterForm({ history }) {
  const styles = useStyles();
  const dispatch = useDispatch();
  const { form, auth, authError, user } = useSelector(({ auth, user }) => ({
    form: auth.register,
    auth: auth.auth,
    authError: auth.authError,
    user: user.user,
  }));

  const [loading, setLoading] = useState(false);

  const onChange = useCallback(
    e => {
      const { value, name } = e.target;
      dispatch(
        changeField({
          form: 'register',
          key: name,
          value,
        }),
      );
    },
    [dispatch],
  );

  const onConfirm = useCallback(() => {
    setLoading(true);
    const { email, password, passwordConfirmation, carType, plateNo } = form;
    let userType = form.userType;
    if (userType === '') {
      userType = 'rider';
    }
    if (!email) {
      window.alert('Email must not be empty.');
    } else if (!password) {
      window.alert('Password must not be empty.');
    } else if (password !== passwordConfirmation) {
      window.alert('Password confirm does not match password.');
    } else if (userType === 'driver' && (carType === '' || plateNo === '')) {
      window.alert('Driver Info must not be empty.');
    } else {
      dispatch(
        register({
          userType,
          email,
          password,
          carType,
          plateNo,
        }),
      );
    }
  }, [dispatch, form]);

  useEffect(() => {
    dispatch(initializeForm('register'));
  }, [dispatch]);

  useEffect(() => {
    if (auth) {
      dispatch(check({ id: auth.id }));
    }
    if (authError) {
      console.log(authError);
    }
  }, [auth, authError, dispatch]);

  useEffect(() => {
    if (user) {
      window.alert('Successfully signed up to Ya-Ta. Welcome!');
      history.push('/login');
    }
  }, [auth, user, history]);

  return (
    <div>
      {loading === true ? (
        <div className={styles.root}>
          <div className={styles.container}>
            <CircularProgress color="secondary" />
          </div>
        </div>
      ) : (
        <Register form={form} onChange={onChange} onClick={onConfirm} />
      )}
    </div>
  );
}

export default withRouter(RegisterForm);
