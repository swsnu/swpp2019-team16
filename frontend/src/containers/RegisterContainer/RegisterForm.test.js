import React from 'react';
import RegisterForm from './RegisterForm';
import { renderWithRedux } from '../../test/utils';
import { MemoryRouter, Route } from 'react-router-dom';

describe('<RegisterForm />', () => {
  const state = {
    auth: {
      register: {
        email: 'email',
        password: 'password',
        passwordConfirmation: 'password',
        vehicleInfo: 'info',
      },
    },
    user: {
      user: null,
    },
  };
  it('SHOULD match with snapshot', async () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/register']}>
        <Route component={RegisterForm} path="/register" />
      </MemoryRouter>,
      state,
    );
    expect(container).toMatchSnapshot();
  });
});
