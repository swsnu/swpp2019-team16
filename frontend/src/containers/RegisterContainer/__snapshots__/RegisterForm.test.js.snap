// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterForm /> SHOULD match with snapshot 1`] = `
<div>
  <div>
    <div
      class="makeStyles-root-3"
    >
      <div
        class="makeStyles-whiteBox-5"
      >
        <div
          class="makeStyles-container-4"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4"
          >
            Join Ya-Ta!
          </h4>
        </div>
        <div
          class="makeStyles-container-4"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            Create Your Account
          </h3>
        </div>
        <div
          class="makeStyles-container-4"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4"
          >
            What are you?
          </h4>
        </div>
        <div
          class="makeStyles-container-4"
        >
          <div
            class="makeStyles-button-6"
          >
            I am a Rider
          </div>
          <div
            class="makeStyles-button-6"
          >
            I am a Driver
          </div>
        </div>
        <div
          class="makeStyles-container-4"
        >
          <div
            class="makeStyles-whiteBox-38"
          >
            <div
              class="makeStyles-container-37"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4"
              >
                Rider Info
              </h4>
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                Email
              </h5>
              <input
                id="email-input"
                name="email"
                type="text"
                value="email"
              />
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                Password
              </h5>
              <input
                id="password-input"
                name="password"
                type="password"
                value="password"
              />
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                Password Confirm
              </h5>
              <input
                id="password-confirmation-input"
                name="passwordConfirmation"
                type="password"
                value="password"
              />
              <div
                class="makeStyles-button-39"
                fullwidth="false"
                variant="contained"
              >
                Confirm
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
